src/ui/public/agg_types/metrics/scripted_metric.js

```diff
@@ -0,0 +1,38 @@
+import AggTypesMetricsMetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import RegistryFieldFormatsProvider from 'ui/registry/field_formats';
+import textHtml from 'ui/agg_types/controls/text.html';
+
+export default function AggTypesMetricsScriptedMetricProvider(Private) {
+  let MetricAggType = Private(AggTypesMetricsMetricAggTypeProvider);
+  let fieldFormats = Private(RegistryFieldFormatsProvider);
+
+  return new MetricAggType({
+    name: 'scripted_metric',
+    title: 'Scripted Metric',
+    makeLabel: function () {
+      return 'Scripted Metric';
+    },
+    getFormat: function () {
+      return fieldFormats.getDefaultInstance('number') || fieldFormats.getDefaultInstance('percent');
+    },
+    params: [
+      {
+        name: 'init_script',
+        type: 'string',
+        editor: textHtml
+      }, {
+        name: 'map_script',
+        type: 'string',
+        editor: textHtml
+      }, {
+        name: 'combine_script',
+        type: 'string',
+        editor: textHtml
+      }, {
+        name: 'reduce_script',
+        type: 'string',
+        editor: textHtml
+      }
```

Comment:
Currently, because of ES, lang can't be passed globally for the 4 scripts.
If someone needs to use another language, a JSON object must be passed instead of a string. 

For example: 
```{
"inline": "....",
"lang": "painless"
}```

Do you want me to add lang selection for each script? 
If so, I think that a selectbox must be also added to switch between file/inline/stored and a new field to hold script params map. It is turning into a heavy development ^^